Docs/Debugging/toolchain.md

```diff
@@ -0,0 +1,105 @@
+# Setting up a local debug toolchain
+
+Most mbed-enabled development boards contain two chips: the target microcontroller and a CMSIS-DAP interface chip. The most common use case of the interface chip is mounting your development board as a USB mass-storage device, making it easy to flash new firmware. However, you can also use this interface chip to debug the target microcontroller without needing an external debugger like a JTAG.
+
+This offers debugging capabilities for stack trace analysis, register dumps and inspection of program execution (breakpoints, watchpoints etc). When combined with a source-level debugger on the development host, such as the GNU Project Debugger (GDB), SWD offers a very rich debugging experience - much more powerful than ``printf()``.
+
+**Tip:** The CMSIS-DAP interface chip does not have to be on the development board. You can also use it as an off-board solution using the [SWDAP debugging probe](https://developer.mbed.org/teams/mbed/wiki/SWDAP). This is a good solution if you're making a custom board, and do not want the added cost (or space required) of an extra interface chip and USB port.
+
+## Running a debug server
+
+To connect to the debug interface, most boards are either supported by [pyOCD](https://github.com/mbedmicro/pyOCD) or [OpenOCD](http://openocd.org). If the debug interface on your board is classified as 'CMSIS-DAP' or 'DAPLink' (most boards) you'll be able to use pyOCD. If not, use OpenOCD.
+
+First, make sure you have the [GNU ARM Embedded Toolchain](https://launchpad.net/gcc-arm-embedded/4.9/4.9-2015-q3-update) installed, and added to your PATH. To verify that you have the right version installed, open a terminal and run:
+
+```
+$ arm-none-eabi-gdb --version
+GNU gdb (GNU Tools for ARM Embedded Processors) 7.8.0.20150604-cvs
+Copyright (C) 2014 Free Software Foundation, Inc.
+License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
+This is free software: you are free to change and redistribute it.
+There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
+and "show warranty" for details.
+```
+
+### pyOCD
+
+If your board is supported by pyOCD, first install [Python 2.7](https://www.python.org/downloads/). Then install pyOCD:
+
+```
+$ pip install pyocd
+```
+
+Then run (with your development board connected over USB):
+
+```
+$ pyocd-gdbserver
+
+INFO:root:DAP SWD MODE initialised
```

Comment:
@AnotherButler This is output by pyOCD, so if it needs to be changed then @c1728p9 would need to do so.